testdata/testcases/docker_image_tests.yml

```diff
@@ -0,0 +1,208 @@
+testSuite:
+  name: Tests all the possible docker image types
+  description: Test all the possible docker image types (DockerImageName|NamedDockerImage|IndexedDockerImage|DockerImageArtifact).
+  tests:
+    - name: DockerImageName
+      description: Test DockerImageName
+      dockerWorkerTaskPayload:
+        command:
+          - echo "Hello world"
+        image: ubuntu
+        maxRunTime: 3600
+      genericWorkerTaskPayload:
+        artifacts: []
+        command:
+          - - bash
+            - '-cx'
+            - >-
+              podman run --name taskcontainer
+              -e "RUN_ID=${RUN_ID}" -e
+              "TASKCLUSTER_ROOT_URL=${TASKCLUSTER_ROOT_URL}" -e
+              "TASKCLUSTER_WORKER_LOCATION=${TASKCLUSTER_WORKER_LOCATION}" -e
+              "TASK_ID=${TASK_ID}"
+              ubuntu 'echo "Hello world"'
+
+              exit_code=$?
+
+              podman rm taskcontainer
+
+              exit "${exit_code}"
+        maxRunTime: 3600
+
+    - name: NamedDockerImage
+      description: Test NamedDockerImage
+      dockerWorkerTaskPayload:
+        command:
+          - echo "Hello world"
+        image:
+          name: ubuntu
+          type: docker-image
+        maxRunTime: 3600
+      genericWorkerTaskPayload:
+        artifacts: []
+        command:
+          - - bash
+            - '-cx'
+            - >-
+              podman run --name taskcontainer
+              -e "RUN_ID=${RUN_ID}" -e
+              "TASKCLUSTER_ROOT_URL=${TASKCLUSTER_ROOT_URL}" -e
+              "TASKCLUSTER_WORKER_LOCATION=${TASKCLUSTER_WORKER_LOCATION}" -e
+              "TASK_ID=${TASK_ID}"
+              ubuntu 'echo "Hello world"'
+
+              exit_code=$?
+
+              podman rm taskcontainer
+
+              exit "${exit_code}"
+        maxRunTime: 3600
+
+    - name: IndexedDockerImage
+      description: Test IndexedDockerImage
+      dockerWorkerTaskPayload:
+        command:
+          - echo "Hello world"
+        image:
+          namespace: test-namespace
+          path: /test/path
+          type: indexed-image
+        maxRunTime: 3600
+      genericWorkerTaskPayload:
+        artifacts: []
+        command:
+          - - bash
+            - '-cx'
+            - >-
+              podman run --name taskcontainer
+              -e "RUN_ID=${RUN_ID}" -e
+              "TASKCLUSTER_ROOT_URL=${TASKCLUSTER_ROOT_URL}" -e
+              "TASKCLUSTER_WORKER_LOCATION=${TASKCLUSTER_WORKER_LOCATION}" -e
+              "TASK_ID=${TASK_ID}"
+              ubuntu 'echo "Hello world"'
```

Comment:
why is it here `ubuntu` if image is coming from namespace?

Reply:
It's not at all.. Sorry, I just left a broken test in here. I'd like it to be fixed once the `IndexedDockerImage` case is fully implemented :)

Nice eye!